File: playwright/PageObjects/PostsPage.js
module.exports = class PostsPage {
  constructor(page) {
    this.page = page;
  }

  async goToCreateNewPost() {
    return this.page.click('a[href="#/editor/post/"]');
  }

  async openPostTypeFilterDropdown() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-selected-item:nth-child(1)');
  }

  async openPostTagsFilterDropdown() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('css=div.gh-contentfilter-menu.gh-contentfilter-tag');
  }

  async selectFilterByPublishedPostsOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Published posts")');
  }

  async selectFilterByScheduledPostsOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Scheduled posts")');
  }

  async selectFilterByDraftedPostsOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Draft posts")');
  }

  async selectFilterByTagName(nameTag) {
    await new Promise(r => setTimeout(r, 1000));
    await this.page.click(`li:text("${nameTag}")`);
  }

  async getFirstPostTitle() {
    await new Promise(r => setTimeout(r, 1000));
    const postsTitles = await this.page.$$(".gh-content-entry-title");
    return postsTitles[0].innerText();
  }

  async clickPostWithTitle(title) {
    return this.page.click(`.gh-content-entry-title:has-text("${title}")`);
  }

  async openPostSortByFilterDropdown() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-selected-item:has-text("Newest")');
  }

  async selectFilterByNewestPostOption() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('.ember-power-select-option:has-text("Newest")');
  }
};

File: kraken/features/web/step_definitions/page_objects/posts_page.rb
class PostsPage
  attr_reader :driver

  def initialize(driver)
    @driver = driver
  end

  def go_to_create_new_post
    @driver.find_element(
      :css, 'a[href="#/editor/post/"]'
    ).click
  end

  def open_post_type_filter_dropdown
    @driver.find_elements(
      :css, '.ember-power-select-selected-item'
    ).first.click
  end

  def open_post_tags_filter_dropdown
    @driver.find_elements(
      :css, '.ember-power-select-selected-item'
    )[2].click
  end

  def select_filter_by_published_posts_option
    @driver.find_elements(
      :css, '.ember-power-select-option'
    ).select do |element|
      element.text == 'Published posts'
    end.first.click
  end

  def select_filter_by_drafted_posts_option
    @driver.find_elements(
      :css, '.ember-power-select-option'
    ).select do |element|
      element.text == 'Draft posts'
    end.first.click
  end

  def select_filter_by_scheduled_posts_option
    @driver.find_elements(
      :css, '.ember-power-select-option'
    ).select do |element|
      element.text == 'Scheduled posts'
    end.first.click
  end

  def select_filter_by_tagname_posts_option(tagName)
    @driver.find_elements(
      :css, '.ember-power-select-option'
    ).select do |element|
      element.text == tagName
    end.first.click
  end

  def click_post_with_title(title)
    @driver.find_elements(
      :css, '.gh-content-entry-title'
    ).select do |element|
      element.text == title
    end.first.click
  end

  def first_post_title
    @driver.find_elements(
      :css, '.gh-content-entry-title'
    ).first.text
  end
end
